src/overlays/actors/ovl_En_Tite/z_en_tite.c

```diff
@@ -39,55 +103,862 @@ const ActorInit En_Tite_InitVars = {
+                               &waterSurfaceY, &waterBox)) &&
+                (this->frontRightFootPos.y <= waterSurfaceY)) {
+                this->frontRightFootPos.y = waterSurfaceY;
+                EffectSsGRipple_Spawn(globalCtx, &this->frontRightFootPos, 0, 220, 0);
+            }
+            thisx->floorPoly = floorPoly;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/func_80B1AA44.s")
+        // If on ground and currently flipped over, set tektite to be fully upside-down
+        if (thisx->bgCheckFlags & 3) {
+            func_800359B8(thisx, thisx->shape.rot.y, &thisx->shape.rot);
+            if (this->flipState >= TEKTITE_FLIPPED) {
+                thisx->shape.rot.z += 0x7FFF;
+            }
+            // Otherwise ensure the tektite is rotating back upright
+        } else {
+            Math_SmoothStepToS(&thisx->shape.rot.x, 0, 1, 1000, 0);
+            if (this->flipState <= TEKTITE_UNFLIPPED) {
+                Math_SmoothStepToS(&thisx->shape.rot.z, 0, 1, 1000, 0);
+                if (thisx->shape.unk_08 > 0) {
+                    thisx->shape.unk_08 -= 400.0f;
+                }
+            }
+        }
+    }
+    thisx->posRot2.pos = thisx->posRot.pos;
+    thisx->posRot2.pos.y += 20.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/func_80B1AA94.s")
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/func_80B1ABBC.s")
+// Get the locations where the tektite feet are rendered
+void EnTite_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** limbDList, Vec3s* rot, void* thisx) {
+    EnTite* this = THIS;
+    switch (limbIndex) {
+        case 8:
+            Matrix_MultVec3f(&sFootOffset, &this->backRightFootPos);
+            break;
+        case 0xD:
+            Matrix_MultVec3f(&sFootOffset, &this->frontRightFootPos);
+            break;
+        case 0x12:
+            Matrix_MultVec3f(&sFootOffset, &this->backLeftFootPos);
+            break;
+        case 0x17:
+            Matrix_MultVec3f(&sFootOffset, &this->frontLeftFootPos);
+            break;
+    }
+    func_80032F54(&this->unk_2C4, limbIndex, 0, 0x18, 0x18, limbDList, -1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/EnTite_Update.s")
+void EnTite_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnTite* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tite/func_80B1B178.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_tite.c", 1704);
+    func_80093D18(globalCtx->state.gfxCtx);
+    func_800628A4(0, &this->collider);
+    if (IS_BLUE) {
+        gSPSegment(__gfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(&D_06001300));
+        gSPSegment(__gfxCtx->polyOpa.p++, 0x09, SEGMENTED_TO_VIRTUAL(&D_06001700));
```

Comment:
POLY_OPA_DISP++
the __gfxCtx is never meant to be used inline, only via the macros